webdriver/tests/bidi/browsing_context/locate_nodes/locator.py

```diff
@@ -0,0 +1,207 @@
+import pytest
+
+from ... import any_string, recursive_compare
+
+
+@pytest.mark.parametrize("type,value", [
+    ("css", "div"),
+    ("xpath", "//div"),
+    ("innerText", "foobarBARbaz")
+])
+@pytest.mark.asyncio
+async def test_find_by_locator(bidi_session, inline, top_context, type, value):
+    url = inline("""<div data-class="one">foobarBARbaz</div><div data-class="two">foobarBARbaz</div>""")
+    await bidi_session.browsing_context.navigate(
+        context=top_context["context"], url=url, wait="complete"
+    )
+
+    result = await bidi_session.browsing_context.locate_nodes(
+        context=top_context["context"],
+        locator={ "type": type, "value": value }
+    )
+
+    expected = [
+        {
+            "type": "node",
+            "sharedId": any_string,
+            "value": {
+                "attributes": {"data-class":"one"},
+                "childNodeCount": 1,
+                "localName": "div",
+                "namespaceURI": "http://www.w3.org/1999/xhtml",
+                "nodeType": 1,
+            }
+        },
+        {
+            "type": "node",
+            "sharedId": any_string,
+            "value": {
+                "attributes": {"data-class":"two"},
+                "childNodeCount": 1,
+                "localName": "div",
+                "namespaceURI": "http://www.w3.org/1999/xhtml",
+                "nodeType": 1,
+            }
+        }
+    ]
+
+    recursive_compare(expected, result["nodes"])
+
+
+@pytest.mark.parametrize("ignore_case,match_type,max_depth,value,expected", [
+    (True, "full", None, "bar", [
+        {
+            "type": "node",
+            "sharedId": any_string,
+            "value": {
+                "attributes": {},
+                "childNodeCount": 1,
+                "children": [],
+                "localName": "strong",
+                "namespaceURI": "http://www.w3.org/1999/xhtml",
+                "nodeType": 1,
+            }
+        },
+        {
+            "type": "node",
+            "sharedId": any_string,
+            "value": {
+                "attributes": {},
+                "childNodeCount": 1,
+                "localName": "span",
+                "namespaceURI": "http://www.w3.org/1999/xhtml",
+                "nodeType": 1,
+            }
+        }]
```

Comment:
NOTE (feel free to ignore): as `recursive_compare` checks the actual results includes expected, but can have extra fields, for the sake of simplicity, the expected values can be simplified to contain only the important for the test fields:
```suggestion
        {
            "type": "node",
            "value": {
                "localName": "strong",
            }
        },
        {
            "type": "node",
            "value": {
                "localName": "span",
            }
        }]
```

Reply:
Did not know that about `recursive_compare`. Nevertheless, I think I'm going to leave these as-is for now, since I want to make sure that the right nodes are serialized, which would include stuff like `childNodeCount`, `attributes`, and so on.